lib/rumoji.rb

```diff
@@ -8,17 +8,25 @@ module Rumoji
 
   # Transform emoji into its cheat-sheet code
   def encode(str)
-    str.gsub(Emoji::ALL_REGEXP) { |match| (emoji = Emoji.find_by_string(match)) && emoji.code || match }
+    str.gsub(Emoji::ALL_REGEXP) do |match|
+      if emoji = Emoji.find_by_string(match)
+        if block_given?
+          yield emoji
+        else
+          emoji.code || match
+        end
+      end
+    end
```

Comment:
The outer if block needs an else clause, that's where the `match` goes (if an emoji is not present return the matched text).